service/src/main/resources/db/migration/V373__calendar_event_time_reservation.sql

```diff
@@ -0,0 +1,57 @@
+CREATE TABLE calendar_event_time
+(
+    id                uuid PRIMARY KEY                  DEFAULT ext.uuid_generate_v1mc(),
+    created_at        timestamp with time zone NOT NULL,
+    created_by        uuid                     NOT NULL REFERENCES evaka_user (id),
+    updated_at        timestamp with time zone NOT NULL,
+    modified_at       timestamp with time zone NOT NULL,
+    modified_by       uuid                     NOT NULL REFERENCES evaka_user (id),
+    calendar_event_id uuid                     NOT NULL REFERENCES calendar_event (id) ON DELETE CASCADE,
+    date              date                     NOT NULL,
+    start_time        time without time zone   NOT NULL,
+    end_time          time without time zone   NOT NULL,
+    child_id          uuid REFERENCES child (id)
+);
+
+CREATE OR REPLACE FUNCTION public.trigger_refresh_updated_at() RETURNS trigger
+    LANGUAGE plpgsql
+AS $$
+BEGIN
+    IF NEW.updated_at = OLD.updated_at THEN
+        NEW.updated_at = NOW();
+    END IF;
```

Comment:
One more small thing: This function should always set `updated_at` and not allow it to be updated from the "outside". That's the whole point of `updated_at`.

Reply:
It seems that `trigger_refresh_updated` is in `R__triggers.sql` so the new function should also be added there I think.

Reply:
> One more small thing: This function should always set `updated_at` and not allow it to be updated from the "outside". That's the whole point of `updated_at`.

Fixed.



> It seems that `trigger_refresh_updated` is in `R__triggers.sql` so the new function should also be added there I think.

Not possible, because repeatable migrations are executed after all pending migrations (`trigger_refresh_updated` is also at V1.sql).

Reply:
Ah, true.